### PR TITLE
Use FUS in AppCenterDistributeV2

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
**Task name**: AppCenterDistributeV2

**Description**: Switch to using FUS for uploading the file.

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Checklist**:
- [ ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [X] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/AppCenterDistributeV2/appcenterdistribute.ts [hunk 1]
```diff
@@ -73,10 +87,10 @@ function responseHandler(defer, err, res, body, handler: () => void) {
     handler();
 }
 
-function beginReleaseUpload(apiServer: string, apiVersion: string, appSlug: string, token: string, userAgent: string): Q.Promise<UploadInfo> {
+function beginReleaseUpload(apiServer: string, apiVersion: string, appSlug: string, token: string, userAgent: string): Q.Promise<any> {
```
Comment (MatkovIvan) [inline Tasks/AppCenterDistributeV2/appcenterdistribute.ts]: Can we avoid any here?

File: Tasks/AppCenterDistributeV2/appcenterdistribute.ts [hunk 2]
```diff
@@ -88,69 +102,77 @@ function beginReleaseUpload(apiServer: string, apiVersion: string, appSlug: stri
     request.post({ url: beginUploadUrl, headers: headers }, (err, res, body) => {
         responseHandler(defer, err, res, body, () => {
             let response = JSON.parse(body);
-            let uploadInfo: UploadInfo = {
-                upload_id: response['upload_id'],
-                upload_url: response['upload_url']
+            if (!response.package_asset_id || (response.statusCode && response.statusCode !== 200)) {
```
Comment (MatkovIvan) [inline Tasks/AppCenterDistributeV2/appcenterdistribute.ts]: All 2xx are valid codes

File: Tasks/AppCenterDistributeV2/appcenterdistribute.ts [hunk 3]
```diff
@@ -369,6 +391,41 @@ function expandSymbolsPaths(symbolsType: string, pattern: string, continueOnErro
...
 
+function getReleaseId(apiServer: string, apiVersion: string, appSlug: string, releaseId: string, token: string, userAgent: string): Q.Promise<any> {
+    tl.debug("-- Getting release id.");
+    let defer = Q.defer<any>();
+    let getReleaseUrl: string = `${apiServer}/${apiVersion}/apps/${appSlug}/uploads/releases/${releaseId}`;
+    tl.debug(`---- url: ${getReleaseUrl}`);
+    let headers = {
+        "X-API-Token": token,
+        "User-Agent": userAgent,
+        "internal-request-source": "VSTS"
```
Comment (MatkovIvan) [inline Tasks/AppCenterDistributeV2/appcenterdistribute.ts:427]: It's still VSTS? (not AzDO)

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: Tasks/AppCenterDistributeV2/appcenterdistribute.ts [hunk 4]
```diff
@@ -88,69 +104,76 @@ function beginReleaseUpload(apiServer: string, apiVersion: string, appSlug: stri
...
         responseHandler(defer, err, res, body, () => {
-            if (body && body['release_url']) {
-                defer.resolve(body['release_url']);
-            } else {
-                defer.reject(tl.loc("FailedToUploadFile"));
+            const { upload_status, message } = body;
+            if (upload_status !== "uploadFinished") {
+                defer.reject(`Failed to patch release upload: ${message}`);
             }
+            defer.resolve(body["release_url"]);
```
Comment (vvechkanov) [inline Tasks/AppCenterDistributeV2/appcenterdistribute.ts]: Do we need to return `release_url` here? We do not use return value later.

File: Tasks/AppCenterDistributeV2/appcenterdistribute.ts (hunk 3 above)
Comment (AnastasiaKubova) [reply Tasks/AppCenterDistributeV2/appcenterdistribute.ts:427]: With this task (AzDO) I receive this error: 
`Cannot find module 'AzDO-task-lib/task' or its corresponding type declarations.`

**Review by Jamminroot — APPROVED** [review]

**Review by annakocheshkova — APPROVED** [review]

**Pull request merged**